### PR TITLE
[Bugfix] Fix QDropout

### PR DESCRIPTION
**Pull request by cenyk1230**
Hi, I find some bugs of my early implementation when using QDropout. 
1. In the backward, the gradient should also be divided by the `1-p` factor. 
2. In the validation step (`self.training = False`), we can directly use the forward of `nn.Dropout` since dropout performs different in training and validation steps. 

Please help check the modification.

### CODE REVIEW

File: actnn/actnn/layers.py [hunk 1]
```diff
@@ -369,13 +369,16 @@ def forward(self, input: torch.Tensor) -> torch.Tensor:
         return ext_quantization.act_quantized_relu(input)
 
 
-class QDropout(nn.Module):
+class QDropout(nn.Dropout):
     def __init__(self, p=0.5):
-        super().__init__()
+        super().__init__(p=p)
         self.p = p
```
Comment (merrymercy) [inline actnn/actnn/layers.py:375]: This line is redundant

**Review by merrymercy — CHANGES_REQUESTED** [review]
